firefox_puppeteer/ui/windows.py

```diff
@@ -0,0 +1,486 @@
+
+        :param property_id: The id to retrieve the value from.
+
+        :returns: The localized string for the requested property.
+
+        :raises MarionetteException: When property id is not found.
+        """
+        return self._l10n.get_localized_entity(self.dtds, property_id)
+
+    def open_window(self, callback=None, expected_window_class=None):
+        """Opens a new top-level chrome window
+
+        :param callback: Optional, function to trigger the window to open. It is
+         triggered with the current :class:`BaseWindow` as parameter.
+         Defaults to `window.open()`
+
+        :param expected_class: Optional, check for the correct window class
+        """
+        # Bug 1121698
+        # For more stable tests register an observer topic first
+        prev_win_count = len(self.marionette.chrome_window_handles)
+
+        self.switch_to()
+        with self.marionette.using_context('chrome'):
+            if callback is not None:
+                callback(self)
+            else:
+                self.marionette.execute_script(""" window.open(); """)
+
+        # Bug 1121698
+        # Observer code should let us ditch this wait code
+        wait = Wait(self.marionette)
+        wait.until(lambda m: len(m.chrome_window_handles) == prev_win_count + 1)
+
+        # Ensure we wait long enough until the window has been fully loaded
+        sleep(.5)
+
+        # Browser windows will always open top-most (with observers we will get the
+        # right handle directly)
+        handle = self._windows.focused_chrome_window_handle
+
+        window = self._windows.create_window_instance(handle, expected_window_class)
+        window.switch_to()
+
+        return window
+
+    def send_shortcut(self, command_key, **kwargs):
+        """Sends a keyboard shortcut to the window
+
+        :param command_key: The key (usually a letter) to be pressed
+
+        :param accel: Optional, If `True` the `Accel` modifier key is pressed. This key
+         differs between OS X (`Meta`) and Linux/Windows (`Ctrl`). Defaults to `False`
+
+        :param alt: Optional, If `True` the `Alt` modifier key is pressed. Defaults to `False`
+
+        :param ctrl: Optional, If `True` the `Ctrl` modifier key is pressed. Defaults to `False`
+
+        :param meta: Optional, If `True` the `Meta` modifier key is pressed. Defaults to `False`
+
+        :param shift: Optional, If `True` the `Shift` modifier key is pressed. Defaults to `False`
+        """
+
+        platform = self.marionette.session_capabilities['platformName'].lower()
+
+        keymap = {
+            'accel': Keys.META if platform is 'darwin' else Keys.CONTROL,
+            'alt': Keys.ALT,
+            'cmd': Keys.COMMAND,
+            'ctrl': Keys.CONTROL,
+            'meta': Keys.META,
+            'shift': Keys.SHIFT,
+        }
+
+        # Append all to press modifier keys
+        keys = []
+        for modifier in kwargs.keys():
```

Comment:
This can just be `modifier in kwargs`, although I would iterate over the keymap, and assert that each kwarg is in the map (I would hate for this to fail silently because I misspelled one of the kwargs).